### ISSUE TITLE
Adding a __hash__ function to buffers and vaos.

### ISSUE THREAD

**Issue by Leterax** [moderngl/moderngl#408]
**A simple use-case of the new feature**

By adding a hash function to the buffer and vao classes they can be used as keys in dicts etc.

**Describe the solution you'd like**

```py
def __hash__(self):
    return hash(self.glo)
```

**Version info**
```
moderngl 5.6.1
--------------
vendor: NVIDIA Corporation
renderer: GeForce GTX 1070/PCIe/SSE2
version: 3.3.0 NVIDIA 446.14
python: 3.7.3 (default, Apr 24 2019, 15:29:51) [MSC v.1915 64 bit (AMD64)]
platform: win32
code: 330

```

**Comment by Leterax** [comment]
easy enough to add for other objects aswell

**Comment by einarf** [comment]
I suspect maybe `id(self)` is better here?

**Comment by einarf** [comment]
Fixed in d559aba
